### PR TITLE
feat: re-export core from cli and update templates

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

- re-export core from cli
- migrate to builtin plugin in create template

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: packages/rspack-plugin-node-polyfill/src/index.js
Comment (h-a-n-a) [inline packages/rspack-plugin-node-polyfill/src/index.js:36]: I was wondering if `compiler.webpack` has the exact same option and outcome  with webpack? Or maybe we should move it to something like `compiler.rspack`

Comment (h-a-n-a) [reply packages/rspack-plugin-node-polyfill/src/index.js:36]: Yes, I think we might have to move `HtmlPlugin` and something like that to `compiler.rspack` to avoid conflict.

**Review by h-a-n-a — APPROVED** [review]

File: packages/rspack-plugin-node-polyfill/src/index.js
Comment (ahabhgk) [reply packages/rspack-plugin-node-polyfill/src/index.js:36]: Then let's not merge this PR and https://github.com/web-infra-dev/rspack-website/pull/445 first, I'd like to have a discuss about this and maybe make some changes, after that then merge the doc PR and introduce builtin plugin to public, just to avoid break change

Comment (h-a-n-a) [reply packages/rspack-plugin-node-polyfill/src/index.js:36]: Sure, let's go!

**Comment by ahabhgk** [comment]
!canary

**Pull request closed without merging**
